ls1intum/Artemis
# `Plagiarism checks`: Add more plagiarism result stats

### Checklist
#### General
#### Server
#### Client

### Motivation and Context
Instructors cannot quickly check basic plagiarism checks statictics. They also don't know who (other instructor or cpc) triggered the job.

### Description
This PR introduces several new statistics to be displayed on the plagiarism results page. They are:
- number of detected submissions
- average similarty
- maximal similarity
- duration of the check (in a simplified form, will close #6143)
- start time
- start by (specific user or the cpc)

Because new DTO is created, this PR includes small refactor of moving other plagiarism related DTOs into a separate package.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exercise

1. Create an exercise.
2. Create two similar submissions for that exercise.
3. Go to the exercise as instructor and open _Check plagiarism_
4. Run the check
5. Verify that stats are correct

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| PlagiarismResultResponseBuilder | 86% | ✅                           |

### Screenshots
New statistics (wide screen):
<img width="1637" alt="image" src="https://github.com/ls1intum/Artemis/assets/32958017/ba73a6e0-ccf5-439c-9c3e-57d1b94b6da4">

New statistics (narrow screen):
<img width="1009" alt="image" src="https://github.com/ls1intum/Artemis/assets/32958017/ea9e536c-25e8-45dd-a0d5-40b62565a667">
